### DISCUSSION TITLE
Azure Private Endpoint and Azure Private Link Service

### DISCUSSION THREAD

**Discussion by yesin25** [AzureCommunityPL/group#452] [Pomoc]
Hello, thank you very much for each comment and your help, I have 2 queries about this please:

1. Although it is possible to deploy a container within a specific Vnet, how would the use of the private link be configured to connect to the Datalake Storage Gen2?

2. Is it necessary that the registry container also be created within the Vnet where the other artifacts are?

[![Architecture.png](https://i.postimg.cc/SNwS5TyZ/Architecture.png)](https://postimg.cc/56SWQmQv)
figure 1. Architecture 1

[![Architecture-Security.png](https://i.postimg.cc/pT05WFQT/Architecture-Security.png)](https://postimg.cc/LY18vhbK)
figure 2.Architecture 2

**Comment by mgrabarz** [comment]
Hi @yesin25,
Welcome to the Polish Azure Community, bit suprised people from Peru posting here :)

1.1 I think networking support for Azure Container Instances (as per icon on your diagram) comes with number of limitations. One of the limitations is Private Endpoint support. Try Container Apps instead, as it should be able to connect to PEs: https://techcommunity.microsoft.com/t5/apps-on-azure-blog/azure-container-apps-virtual-network-integration/ba-p/3096932

1.2 Of course ADLS v2 can be accessed over PE. You need to remember, however, that Storage Accounts come with number of PE types, and one that is needed here is dfs. More here: https://docs.microsoft.com/en-us/azure/storage/common/storage-private-endpoints

1.3 Arow direction is oppsite: ADF should access ADLS, not other way round.

2.1 I would personally keep most of the components private in VNets in most of the enterprise-scale projects. There is however a issue with ACR over PE for Azure DevOps - you didn't say anything about your image build processes, and that may influence the design.

2.2 I also lack clarity about your networking requirements in context of Synapse.
